### PR TITLE
[Enhancement] Use real key type size as the initial agg state size

### PR DESCRIPTION
**Pull request by LiShuMing**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #13880
- Use the real key type size as initial agg state size instead of `16`;
- Split aggregator allocator from aggregator header; 

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by LiShuMing** [comment]
run starrocks_admit_test

File: be/src/exec/vectorized/aggregator.cpp [hunk 1]
```diff
@@ -307,6 +308,32 @@ Status Aggregator::prepare(RuntimeState* state, ObjectPool* pool, RuntimeProfile
     return Status::OK();
 }
 
+void Aggregator::_init_state_allocator(size_t key_type_size) {
```
Comment (stdpain) [inline be/src/exec/vectorized/aggregator.cpp:311]: we should also think about key align

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Comment by LiShuMing** [comment]
run starrocks_be_unittest

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Comment by LiShuMing** [comment]
run starrocks_admit_test

**Pull request closed without merging**